### PR TITLE
fix(scripts): run all clients on main

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-2527

### Changes included:

I've messed up in https://github.com/algolia/api-clients-automation/pull/2789, actually when the base is not main, it means we are already on main. With this addition, every clients will run on main, everytime

### CODE REVIEW

**Comment by shortcuts** [comment]
see PR targetting origin/main https://github.com/algolia/api-clients-automation/actions/runs/9679179736/job/26704621158?pr=3258#step:5:18

see commit on main https://github.com/algolia/api-clients-automation/actions/runs/9680345478/job/26708287046#step:5:18

File: scripts/ci/githubActions/createMatrix.ts [hunk 1]
```diff
@@ -13,8 +13,9 @@ const EMPTY_MATRIX = { client: ['no-run'] };
 
 async function createClientMatrix(baseBranch: string): Promise<void> {
   const matrix: Record<string, ToRunMatrix> = {};
+  // PRs have origin/main as a baseBranch, main is always running from a specific commit
   const commonDependenciesChanged =
-    baseBranch === 'main' || (await isBaseChanged(baseBranch, COMMON_DEPENDENCIES));
+    baseBranch !== 'origin/main' || (await isBaseChanged(baseBranch, COMMON_DEPENDENCIES));
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply scripts/ci/githubActions/createMatrix.ts:18]: yes, maybe a solution is to say if it's not an hash commit? e.g. startswith origin/

[the remark this answers is not in the document]
Comment (shortcuts) [reply scripts/ci/githubActions/createMatrix.ts:18]: I'll create a stack to see if it's true

**Comment by Fluf22** [comment]
Let's try that

**Review by Fluf22 — DISMISSED** [review]

File: scripts/ci/githubActions/createMatrix.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shortcuts) [reply scripts/ci/githubActions/createMatrix.ts:18]: confirmed https://github.com/algolia/api-clients-automation/actions/runs/9680496539/job/26708781501?pr=3261#step:6:1, I can exclude anything not with origin/

**Review by millotp — APPROVED** [review]
looks good !

**Pull request merged**